internal/namespace/test/serialization_test.go

```diff
@@ -0,0 +1,159 @@
+					require.NoError(err)
+					err = os.WriteFile(fmt.Sprintf("testdata/proto/%s-definition-%s.proto", test.schema, objectDef.Name), protoSerialized, 0o644)
+					require.NoError(err)
+
+					vtSerialized, err := objectDef.MarshalVT()
+					require.NoError(err)
+					err = os.WriteFile(fmt.Sprintf("testdata/proto/%s-definition-%s.vtproto", test.schema, objectDef.Name), vtSerialized, 0o644)
+					require.NoError(err)
+				}
+				for _, caveatDef := range compiled.CaveatDefinitions {
+					protoSerialized, err := proto.Marshal(caveatDef)
+					require.NoError(err)
+					err = os.WriteFile(fmt.Sprintf("testdata/proto/%s-caveat-%s.proto", test.schema, caveatDef.Name), protoSerialized, 0o644)
+					require.NoError(err)
+
+					vtSerialized, err := caveatDef.MarshalVT()
+					require.NoError(err)
+					err = os.WriteFile(fmt.Sprintf("testdata/proto/%s-caveat-%s.vtproto", test.schema, caveatDef.Name), vtSerialized, 0o644)
+					require.NoError(err)
+				}
+			} else {
+				files, err := os.ReadDir("testdata/proto")
+				require.NoError(err)
+
+				definitionFiles := map[string]struct{}{}
+				caveatFiles := map[string]struct{}{}
+				for _, file := range files {
+					filename := file.Name()
+					if strings.Contains(filename, test.schema) {
+						// NOTE: this makes some assumptions about the names of the files,
+						// namely that a schema file name will not have either of these
+						// keywords.
+						if strings.Contains(filename, "definition") {
+							definitionFiles[filename] = struct{}{}
+						}
+						if strings.Contains(filename, "caveat") {
+							caveatFiles[filename] = struct{}{}
+						}
+					}
+				}
+
+				vtProtoDefinitions := make(map[string]*core.NamespaceDefinition)
+				standardProtoDefinitions := make(map[string]*core.NamespaceDefinition)
+				definitions := map[string]struct{}{}
+				for definitionFile := range definitionFiles {
+					definition := strings.Split(definitionFile, ".")[0]
+					definitions[definition] = struct{}{}
+					bytes, err := os.ReadFile(fmt.Sprintf("testdata/proto/%s", definitionFile))
+					require.NoError(err)
+
+					standardRepresentation := core.NamespaceDefinition{}
+					err = proto.Unmarshal(bytes, &standardRepresentation)
+					require.NoError(err)
+					standardProtoDefinitions[definitionFile] = &standardRepresentation
+
+					vtRepresentation := core.NamespaceDefinition{}
+					err = vtRepresentation.UnmarshalVT(bytes)
+					require.NoError(err)
+					vtProtoDefinitions[definitionFile] = &vtRepresentation
+				}
+
+				// For each namespace, we want to assert that all of the following are equivalent:
+				// standard serialization -> standard deserialization
+				// vt serialization -> standard deserialization
+				// standard serialization -> vt deserialization
+				// This is to validate that the vt serialization/deserialization isn't doing anything unexpected
+				// compared to the "official" serialization/deserialization
+
+				for definition := range definitions {
```

Comment:
probably can extract this into a helper function and then reuse for caveats below; it can take the definition type in as a generic